rbpf-cli/src/main.rs

```diff
@@ -0,0 +1,241 @@
+}
+
+struct MockSyscall {
+    name: String,
+}
+impl SyscallObject<UserError> for MockSyscall {
+    fn call(
+        &mut self,
+        arg1: u64,
+        arg2: u64,
+        arg3: u64,
+        arg4: u64,
+        arg5: u64,
+        _memory_mapping: &MemoryMapping,
+        result: &mut Result,
+    ) {
+        println!(
+            "Syscall {}: {:#x}, {:#x}, {:#x}, {:#x}, {:#x}",
+            self.name, arg1, arg2, arg3, arg4, arg5,
+        );
+        *result = Result::Ok(0);
+    }
+}
+
+fn main() {
+    let matches = App::new("Solana RBPF CLI")
+        .version("0.2.8")
+        .author("Solana Maintainers <maintainers@solana.foundation>")
+        .about("CLI to test and analyze eBPF programs")
+        .arg(
+            Arg::new("assembler")
+                .about("Assemble and load eBPF executable")
+                .short('a')
+                .long("asm")
+                .value_name("FILE")
+                .takes_value(true)
+                .required_unless_present("elf"),
+        )
+        .arg(
+            Arg::new("elf")
+                .about("Load ELF as eBPF executable")
+                .short('e')
+                .long("elf")
+                .value_name("FILE")
+                .takes_value(true)
+                .required_unless_present("assembler"),
+        )
+        .arg(
+            Arg::new("input")
+                .about("Input for the program to run on")
+                .short('i')
+                .long("input")
+                .value_name("FILE / BYTES")
+                .takes_value(true)
+                .default_value("0"),
+        )
+        .arg(
+            Arg::new("memory")
+                .about("Heap memory for the program to run on")
+                .short('m')
+                .long("mem")
+                .value_name("BYTES")
+                .takes_value(true)
+                .default_value("0"),
+        )
+        .arg(
+            Arg::new("use")
+                .about("Method of execution to use")
+                .short('u')
+                .long("use")
+                .takes_value(true)
+                .possible_values(&["cfg", "disassembler", "interpreter", "jit"])
+                .required(true),
+        )
+        .arg(
+            Arg::new("instruction limit")
+                .about("Limit the number of instructions to execute")
+                .short('l')
+                .long("lim")
+                .takes_value(true)
+                .value_name("COUNT")
+                .default_value(&std::i64::MAX.to_string()),
+        )
+        .arg(
+            Arg::new("trace")
+                .about("Display trace using tracing instrumentation")
+                .short('t')
+                .long("trace"),
+        )
+        .arg(
+            Arg::new("profile")
+                .about("Display profile using tracing instrumentation")
+                .short('p')
+                .long("prof"),
+        )
+        .arg(
+            Arg::new("verify")
```

Comment:
If this is the verifier in the rbpf crate then it won't match the verifier we use on-chain.  I'm going to consolidate these

Reply:
https://github.com/solana-labs/rbpf/pull/177

Reply:
> > > Not all the syscalls are supported, for example the sysvars syacalls probably are not? If so, where does that data come from?
> > 
> > 
> > Right, if run-time needs some information from a blockchain or something like that, it is not available, of course. So what `sysvars` syscall will do in this case, I don't know.
> 
> Be cool if we could send our bpf test programs through the cli to track that it's behavior is inline with the regular runtime. Maybe the cli should run the regular runtime... First question is why can't I run my program that uses sysvars and cpi?

Can we iterate with that in later PRs? It would be good (it seems) to start with something and improve it over time. As is now the tool doesn’t degrade the existing functionality of the cli tool in solana_rbpf repository, but makes it a bit more accessible to users. There will be issues and questions, of course...